system-parachains/bridge-hubs/bridge-hub-polkadot/src/bridge_to_ethereum_unstuck.rs

```diff
@@ -0,0 +1,114 @@
+
+
+pub struct UnstuckSnowbridge;
+
+impl OnRuntimeUpgrade for UnstuckSnowbridge {
+    fn on_runtime_upgrade() -> Weight {
+        if !is_bridge_stuck() {
+            return RocksDbWeight::get().reads(1);
+        }
+
+        log::info!(target: LOG_TARGET, "Updating beacon checkpoint set to unstuck bridge.");
+
+        let checkpoint = checkpoint();
+        CurrentSyncCommittee::<Runtime>::set(sync_committee_keys(checkpoint.current_sync_committee));
+        NextSyncCommittee::<Runtime>::kill();
+        InitialCheckpointRoot::<Runtime>::set(NEW_CHECKPOINT_HEADER_ROOT.into());
+        LatestExecutionState::<Runtime>::kill();
+        FinalizedBeaconStateBuffer::<Runtime>::insert(
+            NEW_CHECKPOINT_HEADER_ROOT.into(),
+            snowbridge_beacon_primitives::CompactBeaconState { slot: checkpoint.header.slot, block_roots_root: checkpoint.block_roots_root },
+        );
+        LatestFinalizedBlockRoot::<Runtime>::set(NEW_CHECKPOINT_HEADER_ROOT.into());
+
+        RocksDbWeight::get().reads_writes(6, 6)
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, TryRuntimeError> {
+        if !is_bridge_stuck() {
+            log::info!(target: LOG_TARGET, "Going to unstuck bridge.");
+        } else {
+            log::info!(target: LOG_TARGET, "Bridge is not stuck.");
+        }
+        Ok(vec![])
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade(_: Vec<u8>) -> Result<(), TryRuntimeError> {
+        frame_support::ensure!(!is_bridge_stuck(), "Bridge is still stuck.");
+        Ok(())
+    }
+}
+
+fn is_bridge_stuck() -> bool {
+    LatestFinalizedBlockRoot::<Runtime>::get() == LAST_IMPORTED_BEACON_HEADER.into()
```

Comment:
We expect that this `LAST_IMPORTED_BEACON_HEADER` will not be reimported, yes?

(I guess the checkpoint is from a block newer than that)